File: content/examples/other/leetcode/problems/valid-palindrome/example-1/solution.js
function isPalindromeFiltered(s) {
  const cleaned = s.toLowerCase().replace(/[^a-z0-9]/g, "");
  let left = 0;
  let right = cleaned.length - 1;
  while (left < right) {
    if (cleaned[left] !== cleaned[right]) return false;
    left += 1;
    right -= 1;
  }
  return true;
}

if (require.main === module) {
  console.log(isPalindromeFiltered("A man, a plan, a canal: Panama"));
  console.log(isPalindromeFiltered("race a car"));
}

module.exports = { isPalindromeFiltered };

File: content/examples/other/leetcode/problems/valid-palindrome/example-1/Solution.java
public class Solution {
  static boolean isPalindromeFiltered(String s) {
    String cleaned = s.toLowerCase().replaceAll("[^a-z0-9]", "");
    int left = 0;
    int right = cleaned.length() - 1;
    while (left < right) {
      if (cleaned.charAt(left) != cleaned.charAt(right)) return false;
      left += 1;
      right -= 1;
    }
    return true;
  }

  public static void main(String[] args) {
    System.out.println(isPalindromeFiltered("A man, a plan, a canal: Panama"));
    System.out.println(isPalindromeFiltered("race a car"));
  }
}
